nautobot/core/testing/views.py

```diff
@@ -113,6 +113,12 @@ def _get_url(self, action, instance=None):
                 return reverse(url_format.format(action), kwargs={"slug": instance.slug})
             except NoReverseMatch:
                 pass
+        # Attempt to resolve using key as the unique identifier if one exists
```

Comment:
Here and the other generic test in `nautobot/core/testing/api.py` it feels like there's a better approach than checking for attrs on the model but I don't know what that approach should be.